pkg/pillar/vault/handler_zfs.go

```diff
@@ -143,6 +143,22 @@ func (h *ZFSHandler) unlockVault(vaultPath string) error {
 
 	// zfs mount
 	if base.IsHVTypeKube() {
+		// Some earlier builds did not have an encrypted etcd vol.
+		// Attempting to load-key on that config will cause entire vault setup to error out.
```

Comment:
How many existing test devices do we have with a TPM?
I wonder whether we can avoid this complexity by just reinstalling them.

Also, does unlockVault() not even run if there is no TPM?

Reply:
I'm not sure of the count but I think it is low, moving to a build off this repo may be a good spot to insist on a reinstall instead of an upgrade.

unlockVault was running due to persist/vault encrypted, but etcd wasn't encrypted from the old image and load-key on that fails.

Reply:
@eriknordmark My latest commit removes the special condition check on etcd and always does a load-key on it if doing a load-key on vault.